rpcs3/Emu/RSX/rsx_cache.h

```diff
@@ -505,39 +513,97 @@ namespace rsx
 			}
 
 			dlg->create();
+			dlg->set_limit(0, entry_count);
+			dlg->set_limit(1, entry_count);
+			dlg->update_msg(0, 0, entry_count);
+			dlg->update_msg(1, 0, entry_count);
+
+			// Setup worker threads
+			unsigned nb_threads = std::thread::hardware_concurrency();
+			std::vector<std::thread> worker_threads(nb_threads);
 
-			const auto prefix_length = version_prefix.length();
-			u32 processed = 0;
-			while (root.read(tmp) && !Emu.IsStopped())
+			// Preload everything needed to compile the shaders
+			// Can probably be parallelized too, but since it's mostly reading files it's probably not worth it
+			std::vector<std::tuple<pipeline_storage_type, RSXVertexProgram, RSXFragmentProgram>> unpackeds;
+			std::chrono::milliseconds last_update;
+			u32 processed_since_last_update = 0;
+			for (u32 i = 0; (i < entry_count) && !Emu.IsStopped(); i++)
```

Comment:
i'd prefer to add blank line before this block. The declaration block above is large enough and has comments anyway